cmd/ipfs/daemon.go

```diff
@@ -423,6 +446,35 @@ func serveHTTPGateway(req cmds.Request) (error, <-chan error) {
 	return nil, errc
 }
 
+type listing struct {
+	Uri  string
+	Keys []string
+}
+
+func loadKeySetFromURLs(urls []string) (key.KeySet, error) {
+	ks := key.NewKeySet()
+	for _, url := range urls {
+		resp, err := http.Get(url)
```

Comment:
![](http://gateway.ipfs.io/ipfs/QmWf5dXQdEQxZgm2shV2KKFTFXoWECCqRE1gavTdJBnKY1/dog-food.jpg)

---

They should be ipfs objects. we can import them from JSON.